new-chatbot-app/src/llm-chain/llm-toolbox/libcal-tools/cancel-reservation/cancel-reservation.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@nestjs/common';
+import { NetworkService } from 'src/shared/services/network/network.service';
+import { Room, LibcalApiAuthorizationService } from '../../../../library-api/libcal-api-auth/libcal-api-auth.service';
+
+@Injectable()
+export class CancelReservationService {
```

Comment:
This class should implement interface LlmTool. And I suggest changing the name to CancelReservationToolService to fit our current naming strategy